### PR TITLE
search cards (title and description)

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob Röhrl <jakob.roehrl@web.de>
* Target version: master 

- [x] case insensitive?
- [ ] search over all boards?

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -44,6 +44,8 @@ Vue.mixin({
 	},
 })
 
+Vue.prototype.$OC = OC
```
Comment (juliushaertl) [inline src/main.js]: We don't need that one.

File: src/main.js [hunk 2]
```diff
@@ -58,5 +60,16 @@ new Vue({
 	el: '#content',
 	router,
 	store,
+	mounted: function() {
+		this.$OC.Search = new OCA.Search(this.filter, this.cleanSearch)
```
Comment (juliushaertl) [inline src/main.js]: no need to assing it anywhere for now, just call `new OCA.Search()`

File: src/store/card.js [hunk 3]
```diff
@@ -30,8 +30,12 @@ export default {
 		cards: [],
 	},
 	getters: {
-		cardsByStack: state => (id) => {
-			return state.cards.filter((card) => card.stackId === id).sort((a, b) => a.order - b.order)
+		cardsByStack: (state, getters) => (id) => {
+			return state.cards.filter(
+				(card) => card.stackId === id
+				&& (card.title.includes(getters.getSearchQuery) || card.description.includes(getters.getSearchQuery))
```
Comment (juliushaertl) [inline src/store/card.js]: Indentation looks a bit odd here, also it would be nicer if we don't do the include calls at all if no search query is set.

**Comment by juliushaertl** [comment]
> case insensitive?

yes, please :wink: 

> search over all boards?

Needs backend, so that is something for a different pr.

File: src/main.js (hunk 2 above)
Comment (jakobroehrl) [reply src/main.js]: If I change 
this.$OC.Search = new OCA.Search(this.filter, this.cleanSearch)
to
this.$OC.Search = new OCA.Search()
the search field disappears.

File: src/main.js (hunk 1 above)
Comment (jakobroehrl) [reply src/main.js]: If I remove it, I get: Error in mounted hook: "TypeError: this.$OC is undefined"

File: src/main.js (hunk 2 above)
Comment (juliushaertl) [reply src/main.js]: Sorry, you got me wrong there. You can just remove the assignment to `this.$OC.Search` and call `new OCA.Search(this.filter, this.cleanSearch)` without assigning the result.

File: src/main.js (hunk 1 above)
Comment (juliushaertl) [reply src/main.js]: Should be fine once the assignment is gone :wink:

**Comment by juliushaertl** [comment]
eslint is still not happy :wink:

**Comment by jakobroehrl** [comment]
>
> eslint is still not happy 😉

/var/www/html/server/apps/deck/src/main.js
  64:3  error  Do not use 'new' for side effects  no-new

/var/www/html/server/apps/deck/src/store/card.js
  35:5  error  Unexpected mix of '&&' and '||'  no-mixed-operators
  36:5  error  Unexpected mix of '&&' and '||'  no-mixed-operators

Mhh, could we merge nevertheless?

**Comment by juliushaertl** [comment]
> Mhh, could we merge nevertheless?

No, if there are false positive errors then we can add an ignore statement, but those two can be handled:

https://eslint.org/docs/rules/no-mixed-operators
https://eslint.org/docs/rules/no-new

**Pull request merged**